.github/workflows/windows-amd64.yml

```diff
@@ -0,0 +1,655 @@
+          path: /tmp/output_graph.*
+      - uses: actions/upload-artifact@v2
+        with:
+          name: "test-checkpoint.${{ matrix.build-flavor }}-${{ matrix.bitrate }}.zip"
+          path: /tmp/checkpoint.tar.xz
+  tensorflow_opt-Windows:
+    name: "Check cache for TensorFlow"
+    runs-on: ubuntu-20.04
+    outputs:
+      status: ${{ steps.check_artifact_exists.outputs.status }}
+      cache_key: ${{ steps.get_cache_key.outputs.key }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - id: get_cache_key
+        uses: ./.github/actions/get_cache_key
+        with:
+          extras: "7"
+      - id: check_artifact_exists
+        uses: ./.github/actions/check_artifact_exists
+        with:
+          name: ${{ steps.get_cache_key.outputs.key }}
+  build-tensorflow-Windows:
+    name: "Build TensorFlow (opt) for Windows"
+    needs: tensorflow_opt-Windows
+    runs-on: windows-2019
+    steps:
+      - run: true
+        if: needs.tensorflow_opt-Windows.outputs.status == 'found'
+      - uses: mozilla/setup-msys2@v2
+        with:
+          msystem: MSYS
+          path-type: inherit
+          update: true
+          install: >-
+            git
+            patch
+            tar
+            unzip
+            zip
+        if: needs.tensorflow_opt-Windows.outputs.status == 'missing'
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.7.9
+        if: needs.tensorflow_opt-Windows.outputs.status == 'missing'
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: 'recursive'
+        if: needs.tensorflow_opt-Windows.outputs.status == 'missing'
+      # It's important that this PATH change only happens *after* the checkout
+      # above, because otherwise the checkout fails when persisisting the
+      # credentials for submodules due to using MSYS2 Git
+      - name: Switch git-bash shell to MSYS2 shell by adding MSYS2 path to PATH front
+        run: echo "D:\a\_temp\msys\msys64\usr\bin" >> $GITHUB_PATH
+        if: needs.tensorflow_opt-Windows.outputs.status == 'missing'
+      - run: ./ci_scripts/tf-setup.sh
+        if: needs.tensorflow_opt-Windows.outputs.status == 'missing'
+      - run: ./ci_scripts/tf-build.sh "--windows-cpu"
+        if: needs.tensorflow_opt-Windows.outputs.status == 'missing'
+      - run: ./ci_scripts/tf-package.sh
+        if: needs.tensorflow_opt-Windows.outputs.status == 'missing'
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ needs.tensorflow_opt-Windows.outputs.cache_key }}
+          path: ${{ github.workspace }}/artifacts/home.tar.xz
+        if: needs.tensorflow_opt-Windows.outputs.status == 'missing'
+  build-lib_Windows:
+    name: "Build libdeepspeech.so and deepspeech C++ binary"
+    runs-on: windows-2019
+    needs: [build-tensorflow-Windows, tensorflow_opt-Windows]
```

Comment:
`build-tensorflow-Windows` already depends on `tensorflow_opt-Windows`, so I don't think you need top repeat it here?

Reply:
We need to repeat it because this job definition needs to access the outputs of the `tensorflow_opt-Windows` job too.